raft/src/main/java/org/apache/kafka/raft/RaftConfig.java

```diff
@@ -138,31 +75,35 @@ public String toString() {
     private final int appendLingerMs;
     private final Map<Integer, InetSocketAddress> voterConnections;
 
-    public RaftConfig(Map<?, ?> props) {
-        this(props, true);
-    }
-
-    protected RaftConfig(Map<?, ?> props, boolean doLog) {
-        super(CONFIG, props, doLog);
-        requestTimeoutMs = getInt(QUORUM_REQUEST_TIMEOUT_MS_CONFIG);
-        retryBackoffMs = getInt(QUORUM_RETRY_BACKOFF_MS_CONFIG);
-        electionTimeoutMs = getInt(QUORUM_ELECTION_TIMEOUT_MS_CONFIG);
-        electionBackoffMaxMs = getInt(QUORUM_ELECTION_BACKOFF_MAX_MS_CONFIG);
-        fetchTimeoutMs = getInt(QUORUM_FETCH_TIMEOUT_MS_CONFIG);
-        appendLingerMs = getInt(QUORUM_LINGER_MS_CONFIG);
-        voterConnections = parseVoterConnections(getList(QUORUM_VOTERS_CONFIG));
-    }
-
-    public static Set<String> configNames() {
-        return CONFIG.names();
-    }
-
-    public static ConfigDef configDef() {
-        return new ConfigDef(CONFIG);
-    }
-
-    public static void main(String[] args) {
-        System.out.println(CONFIG.toHtml());
+    public RaftConfig(AbstractConfig abstractConfig) {
+        this(abstractConfig.getInt(QUORUM_REQUEST_TIMEOUT_MS_CONFIG),
+                abstractConfig.getInt(QUORUM_RETRY_BACKOFF_MS_CONFIG),
+                abstractConfig.getInt(QUORUM_ELECTION_TIMEOUT_MS_CONFIG),
+                abstractConfig.getInt(QUORUM_ELECTION_BACKOFF_MAX_MS_CONFIG),
+                abstractConfig.getInt(QUORUM_FETCH_TIMEOUT_MS_CONFIG),
+                abstractConfig.getInt(QUORUM_LINGER_MS_CONFIG),
+                abstractConfig.getString(QUORUM_VOTERS_CONFIG));
+    }
+
+    public RaftConfig(
+            int requestTimeoutMs,
+            int retryBackoffMs,
+            int electionTimeoutMs,
+            int electionBackoffMaxMs,
+            int fetchTimeoutMs,
+            int appendLingerMs,
+            String votersConnectString
```

Comment:
nit: doesn't matter too much, but could we move this argument to the top to emphasize its importance?

Reply:
Ack.